### PR TITLE
feat: force dashboard

### PR DESCRIPTION
**Pull request by davidruvolo51**
This PR introduces a new vue app for the FORCE project. The purpose of this vue application is to provide a series of interactive visualisations that can be used to explore the dataset. To do this, a dashboard-like UI will be created where all charts can be clicked, and the clicked value updates the overall data query. 

## To Do

- [x] Setup a new vue application and add to the apps config
- [x] Create the initial file structure
- [x] Create a series of charts based on the data structure
- [x] Make all charts interactive and clicked values are added to a filter
- [x] Create a function that renders the filters into graph ql format
- [x] Integrate the filters into all graphql queries
- [x] Create a component that renders clicked values as filter toggles and link with the filter
- [x] Simplify data retrieval and initial transformations (e.g., selecting columns, sorting rows, transforming for pie charts, etc.)
- [x] Fix rendering issue with column charts when filters are removed
- ~~[  ] Connect chart titles and descriptions to selected filters (TBD)~~
- [x] Fix in BarChart: create a new formula for calculating `y` and `dy` when breaking y-axis ticks
- [x] Add typescript
- [x] Apply styling

**Post-review changes**

- [x] Change button label "Update Charts" to "Apply Filters"
- [x] In the BarChart component, when hovering over a bar, there isn't enough space between the bar and the text. Adjust the `dx` to increase the space

## To Test

**This section is in process. Instructions will be more clear when model is integrated into the catalogue** 

For initial testing and review:

1. Navigate to the preview and login
2. Click on schema `FORCE` to view the dashboard

To use the dashboard:

1. Click on one or more of the chart elements (bars, table rows, slices, etc.) to create a new filter. Filters will appear in the "selected filters" bar. Click "Update Charts" to apply the filters
2. To remove a filter, click the minus icon. (the charts will automatically refresh)
3. Click the "Remove all" button to deselect all filters

Things to check:

- Charts respond to filters and the values are correct
- Chart layout is responsive and content isn't compressed
- Layout does not change (too much) when many filters are applied

## Remaining Questions

- [ ] Are the buttons useful?
- [ ] Should charts update automatically when an element is clicked? Or only when the button is clicked? => Morris says: automatic.
- [ ] What titles should be used here?
- [ ] How can we make the functions more generic? I'm guessing it would be good to have a page for the samples table too
- [ ] Should the standard molgenis filters be used here? I feel that this approach is simpler than the emx2 filters.

### CODE REVIEW

**Comment by mswertz** [comment]
Nice. I am curious how we might be able to do this without having to clone the 'app' for each project. But for now happy to go this route so we can learn quickly

**Comment by davidruvolo51** [comment]
I started working on a dashboard configuration file (https://github.com/molgenis/molgenis-emx2/blob/feat/force-dashboard/apps/force/src/config/dashboard.config.json) as I'm building the charts. I think this will allow us to generate pages and content, but it might add layers to post-request data processing.

File: apps/force/src/App.vue [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<template>
+  <Molgenis id="__top" v-model="session">
+    <router-view :session="session" :page="page"></router-view>
+  </Molgenis>
+</template>
+
+<script setup>
```
Comment (chinook25) [inline apps/force/src/App.vue:7]: It would be nice if you'd add typescript when creating new vue files ;)

Comment (mswertz) [reply apps/force/src/App.vue:7]: It was something we agreed upon in a EMX2 dev meet a while a go that we do that default when creating new components.

File: apps/force/src/styles/index.scss [hunk 2]
```diff
@@ -0,0 +1,223 @@
+.app-page,
```
Comment (mswertz) [inline apps/force/src/styles/index.scss:1]: I notice a lot of styling, it looks great and maybe nice way to keep it isolated. However just putting in a comment to ask if we would change again if we move full to tailwind.

File: apps/force/src/styles/palette.json [hunk 3]
```diff
@@ -0,0 +1,70 @@
+{
+  "grays/gray-900": "#0a0a0a",
```
Comment (mswertz) [inline apps/force/src/styles/palette.json:2]: is this tailwind? Maybe I will have my answer later.

**Review by mswertz — APPROVED** [review]

File: apps/force/src/App.vue [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<template>
+  <Molgenis id="__top" v-model="session">
+    <router-view :session="session" :page="page"></router-view>
+  </Molgenis>
+</template>
+
+<script setup lang="ts">
```
Comment (chinook25) [inline apps/force/src/App.vue:7]: I think you also need to add a tsconfig in order for ts to get the highlighting correct in your editor. You can copy one from one of the other apps e.g. molgenis-components.

File: apps/force/index.html [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width,initial-scale=1.0" />
+    <link rel="icon" href="/favicon.ico" />
+    <link rel="stylesheet" href="theme.css" />
+    <title>ERN CRANIO</title>
```
Comment (erikzwart) [inline apps/force/index.html:9]: Rename the title to FORCE dashboard?

**Comment by davidruvolo51** [comment]
Due to merge and rebase conflicts (this branch is long out of date), I'm moving this to a new PR. #3705

**Pull request closed without merging**